src/utilities.ts

```diff
@@ -409,6 +409,11 @@ export function download(query, viewer, options, callback?) {
     var promise = null;
     var m = viewer.addModel();
 
+    if(isRequestProcessing){
```

Comment:
I do not see why you wouldn't want to allow multi-threaded downloads.  Serializing them is not a feature.

Reply:
Ummm...I'm not sure of the use case for downloading the same data multiple times, and users might unintentionally end up performing the same action multiple times due to a lack of visual indication and hence I feel we should throttle the action. Kindly let me know if I'm missing something here.

Reply:
Why do you assume it is the same data? getbin is a general utility.  It is not acceptable for someone to try to fetch two different files at the same time and get an error on the second one.

Reply:
> getbin is a general utility. It is not acceptable for someone to try to fetch two different files at the same time and get an error on the second one.

I 100% agree with this. However, the throttling functionality was added only to the `download` method and not `getbin` so fetching multiple files from `getbin` shouldn't block multiple downloads at once. Anyway, I further double-checked to see if the `download` method is being used elsewhere and I could see that it is used in `example.html` of tests and in several html files of `tests/webpages`. And, it turns out that in `webpages/splitviewer.html` the `download` method is invoked in a loop and throttling would cause a bottleneck here. I had overseen this earlier and hence I was not aware. I shall remove the throttling changes and push a commit. Thank you for brining this to my attention :)

Reply:
More to the point, it is part of the public API so anyone can use it.

Reply:
Right. Going ahead, I'll keep a broader view when proposing changes.